src/routing/Router.ts

```diff
@@ -24,6 +25,9 @@ export interface OutletEvent extends EventObject<string> {
 	action: 'enter' | 'exit';
 }
 
+const ROUTE_SEGMENT_SCORE = 7;
```

Comment:
These numbers look fairly arbitrary at first glance. Perhaps a comment to clarify?

Reply:
They actually are fairly arbitrary, I went for a score that meant we could introduce other types of segments in the future (like maybe wild card, which would have an even smaller score).